src/whir/prover/round_state/state.rs

```diff
@@ -172,22 +172,44 @@ where
                 )
             }
 
-            // Branch: WithStatementSvo - SVO optimization (currently falls back to classic)
+            // Branch: WithStatementSvo - SVO optimization (Algorithm 6 from https://eprint.iacr.org/2025/1117)
             InitialPhase::WithStatementSvo { sumcheck } => {
+                // SVO optimization requirements:
+                // 1. At least 2 * NUM_SVO_ROUNDS (6) variables (NUM_SVO_ROUNDS = 3)
```

Comment:
Just to avoid noting any numbers in case we want to change in the future. Also could be nice to explain why we need this constraint from the paper?

```suggestion
                // 1. At least 2 * NUM_SVO_ROUNDS variables
```